Modules/RatMama.py

```diff
@@ -0,0 +1,206 @@
+
+        if not o2_status != exist_rescue.code_red:
+            diff_response += "O2 Status changed!" if o2_status else\
+                "O2 Status changed, it is now CODE RED!"
+
+        if diff_response:
+            await ctx.reply(diff_response)
+
+    else:
+        platform = None
+
+        if platform_name.casefold() in ("pc", "ps", "xb"):
+            platform = Platforms[platform_name.upper()]
+        else:
+            LOG.warning(f"Got unknown platform from {ctx.user.nickname}: {platform_name}")
+
+        # no case for that name, we have to make our own
+        rescue: Rescue = Rescue(client=client_name, system=system_name, irc_nickname=nickname,
+                                code_red=not o2_status, lang_id=lang_code, platform=platform)
+
+        ctx.bot.board.append(rescue, overwrite=False)
+        index = ctx.bot.board.find_by_name(client=client_name).board_index
+        await ctx.reply(f"RATSIGNAL - CMDR {client_name} - "
+                        f"Reported System: {system_name} (distance to be implemented) - "
+                        f"Platform: {platform_name} - "
+                        f"O2: {'OK' if o2_status else 'NOT OK'} - "
+                        f"Language: {result.group('full_language')}"
+                        f" (Case #{index}) ({platform_name.upper()}_SIGNAL)"
+                        )
+
+
+@rule(r"\bratsignal\b", case_sensitive=False, full_message=True, pass_match=False, prefixless=True)
+async def handle_selfissued_ratsignal(ctx: Context) -> None:
+    """
+     Tries to extract as much details as possible from a self-issued ratsignal and appends
+      these details to the rescue board.
+    Should it be unable to extract the details, it will open a case and ask for the details
+      to be set and will only set the name and nick fields of the rescue.
+
+    Args:
+        ctx: Context of the self-issued ratsignal
+
+    Returns: None
+
+    """
+
+    message: str = ctx.words_eol[0]
+    # the ratsignal is nothing we are interested anymore
+    message = re.sub("ratsignal", "", message, flags=re.I)
+
+    for rescue in ctx.bot.board.rescues.values():
+        if rescue.irc_nickname.casefold() == ctx.user.nickname.casefold():
+            await ctx.reply(
+                "You already sent a signal, please be patient while a dispatch is underway.")
+            return
+
+    sep: str = None
+    if ',' in message:
+        sep = ','
+    elif ';' in message:
+        sep = ';'
+    elif '|' in message:
+        sep = '|'
+    elif '-' in message:
+        sep = '-'
+
+    if not sep:
+        ctx.bot.board.append(Rescue(irc_nickname=ctx.user.nickname, client=ctx.user.nickname))
+        index = ctx.bot.board.find_by_name(ctx.user.nickname)
+        await ctx.reply(f"Case #{index} created for {ctx.user.nickname}, please set details")
+        return
+    parts: List[str] = message.split(sep)
+    system: str = None
+    cr: bool = False
+    platform: Platforms = None
+    for part in parts:
```

Comment:
[minor] seems `parts` is not used elsewhere, may i suggest simplifying this iterator?
```py
for word in message.split(sep):
    ...
```